src/ir/eh-utils.cpp

```diff
@@ -149,9 +149,6 @@ void handleBlockNestedPop(Try* try_, Function* func, Module& wasm) {
 }
 
 void handleBlockNestedPops(Function* func, Module& wasm) {
-  if (!wasm.features.hasExceptionHandling()) {
-    return;
-  }
```

Comment:
I worry this is a significant regression. Before this PR, many passes would skip this pop handling in e.g. normal Emscripten output, just because the feature was not present. That is, this fixes parsing, but it is slowing down optimizations IIUC.

If the parsing issue is that we haven't seen the features section yet, perhaps we can fix that by doing what we now do with names - scan ahead for the features section first?

Reply:
We can do the check in those passes too, as the code here does, I guess - but if that's the best option, perhaps we can avoid the code duplication by adding `handleBlockNestedPopsAfterParse` (or a much better name hopefully) that does the feature check?

Reply:
What do you think about adding an extra optional parameter `bool ignoreFeatures = false` that would only be used by the parser code?

Reply:
Sounds good. Maybe an enum?